node/sync.go

```diff
@@ -1421,6 +1449,34 @@ func (d *Pegnetd) ApplyGradedSPRBlock(tx *sql.Tx, gradedSPRBlock graderStake.Gra
 	return nil
 }
 
+// ApplyGradedDelegatedSPRBlock pays out PEG to the winners of the given GradedBlock.
+// If an error is returned, the sql.Tx should be rolled back by the caller.
+func (d *Pegnetd) ApplyGradedDelegatedSPRBlock(tx *sql.Tx, gradedSPRBlock graderDelegateStake.DelegatedGradedBlock, timestamp time.Time) error {
+	winners := gradedSPRBlock.Winners()
+	for i := range winners {
+		addr, err := factom.NewFAAddress(winners[i].SPR.GetAddress())
+		if err != nil {
+			// TODO: This is kinda an odd case. I think we should just drop the rewards
+			// 		for an invalid address. We can always add back the rewards and they will have
+			//		a higher balance after a change.
+			log.WithError(err).WithFields(log.Fields{
```

Comment:
We should check the address in the code that picks the winners, and we should check the address before we write an entry (in the staking code).  

If the address is already checked before put in the winners list, then you will not have to check here.

Reply:
Yes, Addresses have been checked in staking module writing SPRs.
I believe that double checking is not problem.
What I am concerning here is because of the case when factom library can be broken and factom.NewFAAddress() would return error for correct addresses.
So, it is totally fine to check the address here again.